src/Illuminate/Support/Testing/Fakes/MailFake.php

```diff
@@ -89,14 +137,50 @@ public function hasSent($mailable)
     }
 
     /**
+     * Get all of the queued mailables matching a truth-test callback.
+     *
+     * @param  string  $mailable
+     * @param  callable|null  $callback
+     * @return \Illuminate\Support\Collection
+     */
+    public function queued($mailable, $callback = null)
+    {
+        if (! $this->hasQueued($mailable)) {
+            return collect();
+        }
+
+        $callback = $callback ?: function () {
+            return true;
+        };
+
+        return $this->mailablesOf($mailable, true)->filter(function ($mailable) use ($callback) {
+            return $callback($mailable);
+        });
+    }
+
+    /**
+     * Determine if the given mailable has been queued.
+     *
+     * @param  string  $mailable
+     * @return bool
+     */
+    public function hasQueued($mailable)
+    {
+        return $this->mailablesOf($mailable, true)->count() > 0;
+    }
+
+    /**
      * Get all of the mailed mailables for a given type.
      *
      * @param  string  $type
+     * @param  bool $queued
      * @return \Illuminate\Support\Collection
      */
-    protected function mailablesOf($type)
+    protected function mailablesOf($type, $queued = false)
```

Comment:
Just my two cents but I'd make this a separate method. It would only be duplicating three lines and a method called `queuedMailablesOf` makes the intent much clearer.